File: 1122Programming/leetcode/9. Palindrome Number/ans.cpp
/*
 *
 * https://leetcode.com/problems/palindrome-number/
 * Problem ID: 9. Palindrome Number
 * 
 * Author: Ateto
 * 
 */

class Solution {
public:
    bool isPalindrome(int x) {
        if(x < 0) return false;

        string str;
        while(x) {
            str.push_back('0' + x%10);
            x /= 10;
        }
        for(int i=0; i<str.size()/2; i++) {
            if(str[i] != str[str.size()-1-i]) return false;
        }
        return true;
    }
};

File: 1122Programming/leetcode/9. Palindrome Number/ans.rs
/*
 *
 * https://leetcode.com/problems/palindrome-number/
 * Problem ID: 9. Palindrome Number
 * 
 * Author: revival0728
 * 
 */

impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        let x = x as i64;
        return if x >= 0 {
            let rev_x: i64 = x.to_string().chars().rev().collect::<String>().parse().unwrap();
            rev_x + x == (x << 1)
        } else {
            false
        };
    }
}
